front/lib/api/vaults.ts

```diff
@@ -28,10 +30,21 @@ export const deleteVault = async (
       usages.push(usage.value);
     }
   }
+
+  const dataSources = await DataSourceResource.listByVault(auth, vault);
```

Comment:
there is no magic that will fetch the one from the system vault ? (looking at the code it seems no but just in case)